src/daemon.rs

```diff
@@ -498,28 +547,41 @@ impl Daemon {
             .collect();
         let mut result = vec![];
         for h in self.requests("getblockheader", &params_list)? {
-            result.push(header_from_value(h)?);
+            result.push(header_from_value(h, self.network)?);
         }
         Ok(result)
     }
 
     pub fn getblock(&self, blockhash: &BlockHash) -> Result<Block> {
         let block = block_from_value(
             self.request("getblock", json!([blockhash.to_hex(), /*verbose=*/ false]))?,
+            self.network,
         )?;
+        trace!("latest block {}", block.bitcoin_hash());
         assert_eq!(block.bitcoin_hash(), *blockhash);
         Ok(block)
     }
 
     fn load_blocktxids(&self, blockhash: &BlockHash) -> Result<Vec<Txid>> {
-        self.request("getblock", json!([blockhash.to_hex(), /*verbose=*/ 1]))?
-            .get("tx")
-            .chain_err(|| "block missing txids")?
-            .as_array()
-            .chain_err(|| "invalid block txids")?
-            .iter()
-            .map(parse_hash)
-            .collect::<Result<Vec<Txid>>>()
+        if self.network == Network::Dogecoin || self.network == Network::Dogetest {
+            self.request("getblock", json!([blockhash.to_hex(), /*verbose=*/ true]))?
```

Comment:
Maybe for readability - 
make the `if self.network == Network::Dogecoin || self.network == Network::Dogetest` condition only for the verbose value?...

Reply:
Not sure I can define a variable that will have a different type based on the case in rust. Can I?